src/RecognitionClient.cpp

```diff
@@ -74,6 +75,15 @@ RecognitionClient::~RecognitionClient() = default;
 std::shared_ptr<grpc::Channel> RecognitionClient::createChannel() {
     auto jwt = readFileContent(configuration.getTokenPath());
 
+    if (!configuration.getClientId().empty() && !configuration.getClientSecret().empty()) {
+        auto pair = SpeechCenterCredentials{configuration.getClientId(),
+                                            configuration.getClientSecret(),
+                                            jwt,
+                                            configuration.getTokenPath()}();
+        if (pair.first)
```

Comment:
this kind of implementation makes code very hard to read and understand. There is no real need to optimize this at this level, especially as this is a public example code it should be extremelly verbose